deploy/kubernetes/elastic-agent-kustomize/ksm-hints/elastic-agent-standalone/kustomization.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../../default/elastic-agent-standalone
+
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: DaemonSet
+    metadata:
+      name: elastic-agent-standalone
+      namespace: kube-system
+      labels:
+        app: elastic-agent-standalone
+    spec:
+      template:
+        spec:
+          initContainers:
+            - name: k8s-templates-downloader
+              image: docker.elastic.co/beats/elastic-agent:9.0.0
+              command: ['bash']
+              args:
+                - -c
+                - >-
+                  mkdir -p /usr/share/elastic-agent/state/inputs.d &&
+                  curl -sL https://github.com/elastic/elastic-agent/archive/9.0.tar.gz | tar xz -C /usr/share/elastic-agent/state/inputs.d --strip=5 "elastic-agent-9.0/deploy/kubernetes/elastic-agent-standalone/templates.d"
+              securityContext:
+                runAsUser: 0
+              volumeMounts:
+                - name: elastic-agent-state
+                  mountPath: /usr/share/elastic-agent/state
```

Comment:
Could you extend the integration k8s tests by adding a new one with hints enabled? it would be nice to  check that the agent and all components are starting and reporting healthy with this feature enabled

Reply:
Which tests are you referring to Panos?

Reply:
I had in mind these tests https://github.com/elastic/elastic-agent/blob/main/testing/integration/kubernetes_agent_standalone_test.go#L63 🙂

Reply:
Thank you !! I wanted to double check as we dont actually own them. 

But I created an issue here: https://github.com/elastic/elastic-agent/issues/5660. We can discuss internally for prioritisation

Reply:
AFAICT the k8s ownership is shared, and the agent repo now has k8s integration tests. So ideally anything that has to do with different k8s deployments should be covered by a test but sure discuss https://github.com/elastic/elastic-agent/issues/5660 and try to couple this integration test with this PR